tests/spread/extensions/env-injector/task.yaml

```
@@ -0,0 +1,132 @@
+      echo "Expected: '$env_name=$exp_value', but got: '$env_name=$actual_value'"
+      return 1
+    fi
+
+    return 0 
+  }
+
+  cd "$SNAP_DIR"
+  SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS=1 snapcraft
+
+  unsquashfs "${SNAP}"_1.0_*.snap
+
+
+  # Check that the env-exporter program is present
+  [ -f squashfs-root/bin/command-chain/env-exporter ]
+  
+  # Check that the exec-env script is present
+  [ -f squashfs-root/usr/bin/exec-env ]
+
+  snap install "${SNAP}"_1.0_*.snap --dangerous --classic
+
+  echo "[env-injector] Creating global envfile"
+  touch global.env
+  echo 'HELLO_WORLD="Hello World"' >> global.env
+  global_envpath=$(realpath global.env)
+
+  # Load global envfile
+  snap set env-injector-hello envfile="$global_envpath"
+  # [TEST] - Check if the global envfile is loaded for all apps 
+  check_env_var "env-injector-hello.hello1" "HELLO_WORLD" "Hello World" || exit 1
+  check_env_var "env-injector-hello.hello2" "HELLO_WORLD" "Hello World" || exit 1
+  check_env_var "env-injector-hello.hello-demo" "HELLO_WORLD" "Hello World" || exit 1
+
+  echo "[env-injector] Creating app-specific envfile"
+  touch appenv.env
+  echo 'SCOPED=Scoped' >> appenv.env
+  app_envpath=$(realpath appenv.env)
+
+  # Load app-specific envfile
+  snap set env-injector-hello apps.hello1.envfile="$app_envpath"
+  # [TEST] - Check if the app-specific envfile is loaded for the app
+  check_env_var "env-injector-hello.hello1" "SCOPED" "Scoped" || exit 1
+
+  echo "[env-injector] Setting global env variable"
+  # Set env vars: Global
+  snap set env-injector-hello env.global="World"
+  # [TEST] - Check if the global env var is set for all apps
+  check_env_var "env-injector-hello.hello1" "GLOBAL" "World" || exit 1
+  check_env_var "env-injector-hello.hello2" "GLOBAL" "World" || exit 1
+  check_env_var "env-injector-hello.hello-demo" "GLOBAL" "World" || exit 1
+
+  echo "[env-injector] Setting app-specific env variable"
+  # Set env vars: specific to each app
+  snap set env-injector-hello apps.hello1.env.hello="Hello"
+  snap set env-injector-hello apps.hello2.env.specific="City"
+
+  # [TEST] - Check if the app-specific env var IS SET for the app hello1 
+  check_env_var "env-injector-hello.hello1" "HELLO" "Hello" || exit 1
+  # [TEST] - Check if the app-specific env var IS NOT SET for the app hello2
+  ! check_env_var "env-injector-hello.hello2" "HELLO" "Hello" || exit 1
+
+  # [TEST] - Check if the app-specific env var IS SET for the app hello2
+  check_env_var "env-injector-hello.hello2" "SPECIFIC" "City" || exit 1
+  # [TEST] - Check if the app-specific env var IS NOT SET for the app hello1
+  ! check_env_var "env-injector-hello.hello1" "SPECIFIC" "City" || exit 1
+
+  echo "[env-injector] Rejecting invalid keys"
+  # To check if key with dot will be rejected
+  snap set env-injector-hello env.word.dot="wrong"
+  # [TEST] - Check if the key with dot was rejected
+  ! check_env_var "env-injector-hello.hello1" "DOT" "wrong" || exit 1
```

The test should be to check if the key was "ignored", rather than rejected. The exporter program doesn't reject.

For this test, I'd rather look for "wrong" in the whole env dump to make sure it wasn't exported with an unexpected name, such as WORDDOT, WORD_DOT, WORD. Maybe add a special handling in the function for when the second argument is empty, then call it:
```suggestion
  # [TEST] - Check if the key with dot was ignored
  ! check_env_var "env-injector-hello.hello1" "" "wrong" || exit 1
```

---

Addressed on https://github.com/canonical/snapcraft/pull/4925/commits/979889baf426a72237451834bfe7a3773d565526. Thank you!